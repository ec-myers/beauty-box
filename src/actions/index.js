export const setLipsticks = lipsticks => ({
  type: 'SET_LIPSTICKS',
  lipsticks
})

export const setMascaras = mascaras => ({
  type: 'SET_MASCARAS',
  mascaras
})

export const setFoundations = foundations => ({
  type: 'SET_FOUNDATIONS',
  foundations
})

export const setBlushes = blushes => ({
  type: 'SET_BLUSHES',
  blushes
})

export const setEyeshadows = eyeshadows => ({
  type: 'SET_EYESHADOWS',
  eyeshadows
})

export const setCollection = collection => ({
  type: 'SET_COLLECTION',
  collection
})