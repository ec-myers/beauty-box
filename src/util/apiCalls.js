export const getProduct = async product => {
  const url = `https://makeup-api.herokuapp.com/api/v1/products.json?product_type=${product}`;
  const response = await fetch(url);
  const data = await response.json();
  const splicedData = data.splice(0, 20);

  const cleanedData = cleanProductData(splicedData);
  if (!response.ok) {
    throw new Error('Could not fetch product');
  }
  return await Promise.all(cleanedData);
}

const cleanProductData = (data) => {
  return data.map(async result => {
    const {id, brand, name, image_link, description, tag_list} = result;
    return { id, brand, name, image_link, description, tag_list}
  });
};