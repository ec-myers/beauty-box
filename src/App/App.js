import React, { Component } from 'react';
import './App.scss';
import { getProduct } from '../util/apiCalls';
import { setLipsticks, setMascaras } from '../actions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

class App extends Component {
  componentDidMount = async() => {
    const { setLipsticks, setMascaras } = this.props;
    try{
      const lipsticks = await getProduct('lipstick');
      setLipsticks(lipsticks);
      const mascaras = await getProduct('mascara');
      setMascaras(mascaras);

    } catch ({message}){
      console.log(message)
    }
  }

  render() {
    return(
      <p>Beauty Box</p>
    )
  }
}

export const mapStateToProps = ({ lipsticks, mascaras }) => ({
  lipsticks,
  mascaras
});

export const mapDispatchToProps = dispatch => {
  return bindActionCreators({ setLipsticks, setMascaras }, dispatch) 
};

export default connect(mapStateToProps, mapDispatchToProps)(App);
