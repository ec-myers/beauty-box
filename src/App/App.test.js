import React from 'react';
import { shallow } from 'enzyme';
import { App, mapStateToProps, mapDispatchToProps } from './App';
import * as actions from '../actions/index';
import { getProduct } from '../util/apiCalls';

jest.mock('../util/apiCalls.js');
jest.mock('../actions');

describe('App', () => {
  let lipsticks = [{}, {}, {}]
  let mascaras = [{}, {}, {}]
  let foundations = [{}, {}, {}]
  let blushes = [{}, {}, {}]
  let eyeshadows = [{}, {}, {}]
  let allCategories=[{}, {}, {}]
  let wrapper = shallow(<App 
    setLipsticks={actions.setLipsticks}
    setMascaras={actions.setMascaras}
    setFoundations={actions.setFoundations}
    setBlushes={actions.setBlushes}
    setEyeshadows={actions.setEyeshadows}
    setCollection={actions.setCollection}
    setError={actions.setError}
    setLoading={actions.setLoading}
    allCategories={allCategories}
    lipsticks={lipsticks}
    mascaras={mascaras}
    foundations={foundations}
    blushes={blushes}
    eyeshadows={eyeshadows}
  />);

 it('should match the snapshot', () => {

  expect(wrapper).toMatchSnapshot();
 });

 it('should call setLipsticks', () => {
   
  expect(actions.setLipsticks).toHaveBeenCalled();
 });

 it('should add a product to collection', () => {
  let newCollection = [{}, {}];
  let mockAction = 

  expect(setCollection(newCollection)).toEqual(newCollection);
 });
});

describe('mapStateToProps', () => {
  it('should return an object with 8 properties', () => {
    let lipsticks, mascaras, foundations, blushes, eyeshadows, collection, error, isLoading;
    let mockState = {
      lipsticks,
      mascaras,
      foundations,
      blushes,
      eyeshadows,
      collection,
      error,
      isLoading
    };

    let expected = {
      lipsticks,
      mascaras,
      foundations,
      blushes,
      eyeshadows,
      collection,
      error,
      isLoading
    };

    const mappedProps = mapStateToProps(mockState);

    expect(mappedProps).toEqual(expected);
  });
});

describe('mapDispatchToProps', () => {
  let mockDispatch;

  beforeEach(() => {
    mockDispatch = jest.fn();
  });

  it('should call dispatch with a setLipsticks action', () => {
    const actionToDispatch = actions.setLipsticks([{}, {}, {}]);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setLipsticks([{}, {}, {}]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should call dispatch with a setMascaras action', () => {
    const actionToDispatch = actions.setMascaras([{}, {}, {}]);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setMascaras([{}, {}, {}]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should call dispatch with a setFoundations action', () => {
    const actionToDispatch = actions.setFoundations([{}, {}, {}]);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setFoundations([{}, {}, {}]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should call dispatch with a setBlushes action', () => {
    const actionToDispatch = actions.setBlushes([{}, {}, {}]);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setBlushes([{}, {}, {}]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should call dispatch with a setEyeshadows action', () => {
    const actionToDispatch = actions.setEyeshadows([{}, {}, {}]);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setEyeshadows([{}, {}, {}]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should call dispatch with a setCollection action', () => {
    const actionToDispatch = actions.setCollection([{}, {}, {}]);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setCollection([{}, {}, {}]);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should call dispatch with a setError action', () => {
    const actionToDispatch = actions.setError('Error');
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setError('Error');

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should call dispatch with a setLoading action', () => {
    const actionToDispatch = actions.setLoading(true);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.setLoading(true);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
