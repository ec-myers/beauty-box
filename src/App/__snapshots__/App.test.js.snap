// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<Fragment>
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/products/:type"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/collection"
      render={[Function]}
    />
    <Route
      exact={true}
      path="/shopall"
      render={[Function]}
    />
    <Route
      component={[Function]}
    />
  </Switch>
</Fragment>
`;
